### PR TITLE
Add the frontend for Jax random Truncated normal

### LINKED ISSUE #14882: truncated_normal

### PR DESCRIPTION
**Pull request by iammarco11**
closes #14882

### CODE REVIEW

**Comment by iammarco11** [comment]
Hi @hmahmood24 
The latest commit test fails and outputs the following
```
=================================== FAILURES ===================================
_ test_jax_truncated_normal[cpu-ivy.functional.backends.tensorflow-False-False] _
ivy/utils/exceptions.py:122: in _handle_exceptions
    return fn(*args, **kwargs)
ivy/functional/ivy/data_type.py:923: in as_ivy_dtype
    return current_backend(None).as_ivy_dtype(dtype_in)
ivy/functional/backends/tensorflow/data_type.py:202: in as_ivy_dtype
    dtype_str = ivy_dtype_dict[dtype_in]
E   TypeError: unhashable type: 'DeviceArray'
```

Now I am not sure why this happens, I researched a bit and came across this [issue](https://github.com/google/jax/issues/6233) and came across something called `HashableArrayWrapper`.
Is the backend faulty, or is it something to do with the way I implemented the frontend. Looking forward to hear your inputs and thoughts :)

File: ivy/functional/frontends/jax/random.py [hunk 1]
```diff
@@ -171,3 +172,19 @@ def t(key, df, shape=(), dtype="float64"):
 def randint(key, shape, minval, maxval, dtype="int64"):
     seed = _get_seed(key)
     return ivy.randint(minval, maxval, shape=shape, dtype=dtype, seed=seed)
+
+
```
Comment (hmahmood24) [inline ivy/functional/frontends/jax/random.py:176]: Kindly add the `@to_ivy_arrays_and_back` and `@handle_jax_dtype` decorators to the function here

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @iammarco11, thanks so much for opening this PR! 😃 I have made a comment on the PR. Can you kindly address that and then try running the tests to see if they pass or still throw the same error. Let me know if you need any help afterwards. Thanks!

**Comment by iammarco11** [comment]
Hi @hmahmood24 I did the changes you requested but the same error persists. Could you help me out? Thanks for taking a look into it!

**Comment by iammarco11** [comment]
@hmahmood24 fixed the test scenario. Finished implementing the truncated normal for jax frontend. This PR is now ready to be reviewed. Looking forward to hearing your inputs. Thanks!

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_random.py [hunk 2]
```diff
@@ -125,6 +125,65 @@ def call():
         assert u.shape == v.shape
 
 
+# TODO Update the test by fixing the uint32 unsupported problem
+@pytest.mark.xfail
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_random.py:129]: Can you elaborate on why you've used this here?

**Review by hmahmood24 — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by hmahmood24** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**